pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractBaseDispatcher.java

```diff
@@ -227,6 +236,15 @@ && trackDelayedDelivery(entry.getLedgerId(), entry.getEntryId(), msgMetadata)) {
                 ((AbstractTopic) topic).addFilteredEntriesCount(filtered);
             }
         }
+        if (CollectionUtils.isNotEmpty(entriesToRedeliver)) {
+            this.subscription.getTopic().getBrokerService().getPulsar().getExecutor()
+                    .schedule(() -> {
+                        // simulate the Consumer rejected the message
+                        subscription
+                                .redeliverUnacknowledgedMessages(consumer, entriesToRedeliver);
+                    }, 1, TimeUnit.SECONDS);
+
+        }
```

Comment:
Need to check the subscription type here? The Single active consumer subscription mode will perform a different behavior.

Reply:
with "Single active consume" the outcome it the same, in that case we are deferring the processing of the message (like when the consumer negatively acks the message on the client because the message cannot currently be processed due to some reason).

Unfortunately the EntryFilter cannot return a value for the delay period without breaking the API (we only have a enum)

Reply:
But for a single active consumer, it will rewind the cursor multiple times. It's not like Shared subscription(only redeliver the given message)